klaus/__init__.py

```diff
@@ -17,8 +18,9 @@ class Klaus(flask.Flask):
 
     def __init__(self, repo_paths, site_name, use_smarthttp, ctags_policy='none'):
         """(See `make_app` for parameter descriptions.)"""
-        repo_objs = [FancyRepo(path) for path in repo_paths]
+        repo_objs, invalid_repos = self.load_repos(repo_paths)
         self.repos = dict((repo.name, repo) for repo in repo_objs)
+        self.invalid_repos = invalid_repos
```

Comment:
Shouldn't we also use `.name` for invalid repos? It's based on the path only, so you could factor it out from `Repo.name` to e.g. `utils.something` and use it here as well.

Reply:
I am not sure what you mean here. Are you suggesting something like `self.invalid_repos = dict((repo.name, repo) for repo in invalid_repos)`? I don't see the use of a dict here...

Reply:
True, at the moment we don’t need it since we’re not mapping invalid repos to urls. If we do that at some point we can change it to a dict. 

OTOH we could even stuff both types of repo objects into the same dict and have a check whenever we list or show repos.

Reply:
I would rather keep two separated collections. I can't think of an operation we would want to apply to both valid and invalid repositories without distinction (even for display, we always want to separate valid and invalid repositories...). Having them in the same collection would requires to split the collection whenever we iterate through it, which is error prone.